content/200-concepts/100-components/02-prisma-client/260-client-extensions/100-client-extensions-model.mdx

```diff
@@ -0,0 +1,135 @@
+
+Possible uses include the following:
+
+- New operations to operate alongside existing Prisma Client operations, such as `findMany`
+- Encapsulated business logic
+- Repetitive operations
+- Model-specific utilities
+
+You can add methods to a specific model in your schema or to all models in your schema.
+
+To create Prisma Client extensions, use the [`$extends`](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions) client-level method. To add custom methods to your models, use the `model` extension type.
+
+</TopBlock>
+
+## Add the <inlinecode>clientExtensions</inlinecode> feature flag to <inlinecode>schema.prisma</inlinecode>
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the generator block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add a custom method to a specific model
+
+To extend a specific model in your schema, use the following structure. This example adds a method to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: '<name>',
+  model?: {
+    user: { ... }   // in this case, we extend the `user` model
+  },
+});
+```
+
+- `name`: an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+The following example adds a method called `signUp` to the `user` model. This method creates a new user with the specified email address.
+
+```ts
+const xprisma = prisma.$extends({
+  model: {
+    user: {
+      async signUp(email: string) {
+        await prisma.user.create(...)
+      },
+    },
+  }
+})
+```
+
+You would call `signUp` in your application as follows:
+
+```ts
+const user = await xprisma.user.signUp('john@prisma.io')
+```
+
+<Admonition type = "info">
+
+When you call a method in your extension, you must use the correct instance in your code. In the above example, `xprisma.user.signUp` works, but `prisma.user.signUp` does not.
+
+</Admonition>
+
+## Add a custom method to all models in your schema
+
+To extend all models in your schema, use the following structure:
+
+```ts
+const xprisma = prisma.$extends({
+  name?: '<name>',
+  model?: {
+    $allModels: { ... }
+  },
+})
+```
+
+- `name` is an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+The following example adds a `getClass` method to all models.
+
+```ts
+const xprisma = new prisma.$extends({
+  model: {
+    $allModels: {
+      getClass<T extends object>(this: T): new () => T {
+        return class {
+          /** Generic Implementation */
+        } as any
+      },
+    },
+  },
+})
+
+class UserService extends xprisma.user.getClass() {
+  method() {
+    const user = this.findFirst({}) // fully type-safe
+  }
+}
```

Comment:
Uh, what does this do? I can not easily understand just from the code.

Reply:
@SevInf  Could you provide some commentary on this code block, so we can include an explanation in the docs? Or can we provide a simpler example?

Reply:
I'd use something other than `getClass` example for model extensions, like validation. `getClass` thing is mostly a workaround for working with one specific JS framework and won't be that useful to other readers. I am pretty sure it will be pretty confusing for them as well.

Reply:
Thanks @SevInf  - I've added the example you provided in your other comment